### DISCUSSION TITLE
Does vcpkg have support for `OBJECT` type libraries?

### DISCUSSION THREAD

**Discussion by bwrsandman** [microsoft/vcpkg#36012] [Q&A]
I understand that ImGui is intended for direct inclusion in projects rather than static linking, a point I know from having [asked the author directly on Twitter](https://twitter.com/ocornut/status/1169388400287174656). However, this method presents challenges, particularly in managing a third_party/ directory, which can be difficult to maintain and update.

The rationale for incorporating ImGui directly relates to its customization #defines, complicating the development of a one-size-fits-all library. Upon exploring [OBJECT type libraries in CMake](https://cmake.org/cmake/help/latest/command/add_library.html#object-libraries), I've found that they might provide a suitable middle ground. These libraries do not pre-compile, allowing ImGui's #defines to stay unresolved until a project integrates them. This could facilitate late customization via an imgui_config.h or CMake definitions.

I'm wondering if there is already a vcpkg package exported as an OBJECT type, and whether this would be an effective approach for projects like ImGui.

**Comment by Neumann-A** [comment]
> These libraries do not pre-compile

This is incorrect. OBJECT targets are compiled.

**Reply by bwrsandman** [reply]
Ahhh you're right! So it's the linking that's done later?
Is there any source-group type construct like what I was describing that exists in cmake?
